### PR TITLE
ikev1: add ikev1 parser - v12

### PR DESCRIPTION
**Pull request by fhonza**
Previous PR: #5512 

Describe changes:
* add DETECT_UINT_RA (range) to keyword documentation
* rustfmt on ike files
* small updates to src/output-json-ike.c 

suricata-verify-pr: 351

### CODE REVIEW

**Comment by catenacyber** [comment]
Did you run `git mv src/output-json-ikev2.h src/detect-ike-vendor.h ` ? Github seems to think so

**Comment by catenacyber** [comment]
Could you rebase on top of master with commit d2c8c9f58e6608521afaf4484292d607e435f8c4 to get CI going ?

**Comment by fhonza** [comment]
> Could you rebase on top of master with commit [d2c8c9f](https://github.com/OISF/suricata/commit/d2c8c9f58e6608521afaf4484292d607e435f8c4) to get CI going ?

yes, the force push only contains the rebase, hope for the best :)

File: rust/src/ike/parser.rs [hunk 1]
```diff
@@ -0,0 +1,739 @@
...
+}
+
+pub fn parse_security_association(i: &[u8]) -> IResult<&[u8], SecurityAssociationPayload> {
+    do_parse!(
+        i,
+        domain_of_interpretation: be_u32
+            >> situation: cond!(domain_of_interpretation == 1, take!(4))
+            >> data: cond!(
+                domain_of_interpretation == 1 && i.len() >= 8,
+                take!(i.len() - 8)
```
Comment (catenacyber) [inline rust/src/ike/parser.rs:286]: `rest` seems simpler, no ?

**Comment by fhonza** [comment]
> Did you run `git mv src/output-json-ikev2.h src/detect-ike-vendor.h ` ? Github seems to think so

Strange, no certainly not. checked locally with tig, have only src/{output-json-ikev2.h => output-json-ike.h} here.
Perhaps some github-heuristic which tries to guess things?

**Review by catenacyber — COMMENTED** [review]
I would like to see the first commits merged already cf https://github.com/OISF/suricata/pull/5512#issuecomment-720999311

File: src/output-json-ike.c [hunk 2]
```diff
@@ -60,33 +63,41 @@ typedef struct LogIKELogThread_ {
     MemBuffer *buffer;
 } LogIKELogThread;
 
+bool EveIKEAddMetadata(const Flow *f, uint64_t tx_id, JsonBuilder *js)
```
[the remark this answers is not in the document]
Comment (fhonza) [reply src/output-json-ike.c:66]: Yes, protocol keywords were added for ikev1 and some of them can be applied for ikev2 as well. Previously there were no keywords for ikev2.

File: src/output-json-ike.c [hunk 3]
```diff
@@ -60,33 +63,41 @@ typedef struct LogIKELogThread_ {
...
+}
+
 static int JsonIKELogger(ThreadVars *tv, void *thread_data, const Packet *p, Flow *f, void *state,
         void *tx, uint64_t tx_id)
 {
-    IKETransaction *iketx = tx;
     LogIKELogThread *thread = thread_data;
-
-    JsonBuilder *jb = CreateEveHeader((Packet *)p, LOG_DIR_PACKET, "ike", NULL);
-    if (unlikely(jb == NULL)) {
```
[the remark this answers is not in the document]
Comment (fhonza) [reply src/output-json-ike.c]: honestly i can't remember, i can change it back....

File: src/output-json-ike.c (hunk 2 above)
[the remark this answers is not in the document]
Comment (catenacyber) [reply src/output-json-ike.c:66]: This is not about protocol keywords
This is about logging IKE (v1 or v2) data with alerts.
This is a new feature you are adding :-)

File: src/output-json-ike.c (hunk 3 above)
[the remark this answers is not in the document]
Comment (catenacyber) [reply src/output-json-ike.c]: Let's minimize the changes :-)

File: src/output-json-ike.c (hunk 2 above)
[the remark this answers is not in the document]
Comment (fhonza) [reply src/output-json-ike.c:66]: yes i know, i've added the logging of metadata for alerts which are triggered by the keywords :)

[the remark this answers is not in the document]
Comment (catenacyber) [reply src/output-json-ike.c:66]: No matter how they are triggered :-)

[the remark this answers is not in the document]
Comment (fhonza) [reply src/output-json-ike.c:66]: ...yes ok....let's say it was my initial motivation for that :P

**Comment by satta** [comment]
Is there anything left to do here for a new PR other than
- the 'nit' empty line removal, and
- the adjustment of the date in the comments?

I might be willing to do some work here to move things forward but don't have the full context -- sorry.

**Comment by catenacyber** [comment]
New PR is welcome :-)

The `EveIKEAddMetadata` adding should be in its own commit as well

I think I did not review yet the keywords logic versus the IKE version...

@victorjulien does it make sense to split this PR in 2 : one PR for the commits preparing this (like detect: add comparison-mode LTE/GTE for Detect(U32/u8)Data), one PR for the big IKEv2 changes

**Comment by satta** [comment]
> The `EveIKEAddMetadata` adding should be in its own commit as well

OK.

[...]
> @victorjulien does it make sense to split this PR in 2 : one PR for the commits preparing this (like detect: add comparison-mode LTE/GTE for Detect(U32/u8)Data), one PR for the big IKEv2 changes

I'll prepare a new PR once this question is sorted out :)

File: src/output-json-ike.h [hunk 4]
```diff
@@ -1,4 +1,4 @@
-/* Copyright (C) 2015 Open Information Security Foundation
+/* Copyright (C) 2020 Open Information Security Foundation
```
[the remark this answers is not in the document]
Comment (satta) [reply src/output-json-ike.h:1]: Done in next PR.

File: src/output-json-ike.c (hunk 3 above)
[the remark this answers is not in the document]
Comment (satta) [reply src/output-json-ike.c]: Changed back in next PR to minimize changes.

**Comment by satta** [comment]
New PR: #5697

**Pull request closed without merging**